text/2018-08-29-destroy-range.md

```diff
@@ -0,0 +1,50 @@
+# Summary
+
+Support RPC `DestroyRange`. This call is on the whole TiKV rather than a certain Region. When it is invoked, TiKV will use `delete_files_in_range` to quickly free a large amount of space, and then scan and delete all remaining keys in the range. Raft layer will be bypassed, and the range can cover across multiple Regions. The invoker should promise that after invoking `DestroyRange`, the range will never be accessed again. That is to say, the range is permanently scrapped.
+
+This interface is only designed for TiDB. It's used to clean up data after dropping/truncating a huge table/index.
+
+# Motivation
+
+Currently, when TiDB drops/truncates a table/index, after GC lifetime it will invoke `DeleteRange` on all affected Regions respectively. Though TiKV will call `delete_files_in_range` to quickly free up the space for each Region, the SST files that span over multiple Regions cannot be removed immediately. As a result, the disk space is not able to be recycled in time. Another problem is that the process to delete all Regions respectively is too slow.
+
+This issue has troubled some of our users. It should be fixed as soon as possible.
+
+# Detailed design
+
+1. Add related protobuf for `DestroyRange` to kvrpcpb.
+    ```protobuf
+    message DestroyRangeRequest {
+        Context context = 1;
+        bytes start_key = 2;
+        bytes end_key = 3;
+    }
+    ```
+    The `context` in the request body does not indicate which Region to work. Fields about Region, Peer, etc. are ignored.
```

Comment:
so why must we take a Context? any field do we need? you should mention it here

Reply:
Just to keep uniform with other requests. Also context is used to pass some parameters like `fill_cache`. It's possible that later there will be more parameters in the context which is needed by destroy_range.